### PR TITLE
[FIXED JENKINS-39290] A debug flag to disable NIO

### PR DESCRIPTION
**Pull request by kohsuke**
To further isolate problems during remote trouble-shooting of remoting, define a switch to bypass NIO and use classic blocking I/O transport.

Ample warning to prevent accidental dependencies from other Java code.

### CODE REVIEW

**Comment by kohsuke** [comment]
(For my tracking purpose, this is CJP-5822)

**Comment by kohsuke** [comment]
@reviewbybees

**Comment by oleg-nenashev** [comment]
For OSS tracking purpose, please create a JIRA issue.
The new flag should be also documented in "doc/configuration.md"

File: src/main/java/org/jenkinsci/remoting/nio/NioChannelHub.java [hunk 1]
```diff
@@ -699,4 +699,15 @@ public void ensureValid() throws IOException {
...
+    /**
+     * Escape hatch to disable NIO-based remoting.
+     *
+     * @since 2.26.3
+     * @deprecated
+     *      This field is a debug switch and not a part of the committed interface of this library,
+     *      but needs to be public and mutable for example so that it can be set from script console.
+     *      Subject to change without notice. Do not rely on this from application code.
+     */
+    public static boolean DISABLE_NIO = Boolean.getBoolean(NioChannelHub.class.getName()+".disabled");
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/remoting/nio/NioChannelHub.java]: Should be <code>@Restricted</code> or final

**Comment by kohsuke** [comment]
Changed as you wish

**Comment by amuniz** [comment]
:bee:

**Pull request merged**